src/main/java/ru/skypro/homework/controller/AdsController.java

```diff
@@ -0,0 +1,276 @@
+package ru.skypro.homework.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import ru.skypro.homework.dto.AdDto;
+import ru.skypro.homework.dto.AdsDto;
+import ru.skypro.homework.dto.CreateOrUpdateAdDto;
+import ru.skypro.homework.dto.ExtendedAdDto;
+
+import javax.validation.Valid;
+import java.util.Collection;
+
+@Slf4j
+@CrossOrigin(value = "http://localhost:3000")
+@RestController
+@RequiredArgsConstructor
+@Tag(name = "Ads controller", description = "объявления")
+@RequestMapping("ads")
+public class AdsController {
+
+    @Operation(
+            summary = "Получение всех объявлений",
+            responses = {
+                    @ApiResponse(
+                            responseCode = "200",
+                            description = "OK",
+                            content = {
+                                    @Content(
+                                            mediaType = MediaType.APPLICATION_JSON_VALUE,
+                                            schema = @Schema(implementation = AdsDto.class)
+                                    )
+                            }
+                    )
+            }
+    )
+
+    @GetMapping()
+    public ResponseEntity<Collection<AdDto>> getAllAds() {
+        return null;
+    }
+
+    @Operation(
+            summary = "Добавление объявления",
+            responses = {
+                    @ApiResponse(
+                            responseCode = "201",
+                            description = "Created",
+                            content = {
+                                    @Content(
+                                            mediaType = MediaType.APPLICATION_JSON_VALUE,
+                                            schema = @Schema(implementation = AdDto.class)
+                                    )
+                            }
+                    ),
+                    @ApiResponse(
+                            responseCode = "401",
+                            description = "Unauthorized",
+                            content = {
+                                    @Content(
+                                    )
+                            }
+                    )
+            }
+    )
+
+    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity<AdDto> addAd(@RequestBody @Valid AdDto properties, @RequestBody MultipartFile image) {
```

Comment:
```suggestion
    public ResponseEntity<AdDto> addAd(@RequestPart @Valid AdDto properties, @RequestPart MultipartFile image) {
```
аннотации неправильные, должны быть RequestPart

Reply:
подскажи пожалуйста, @RequestPart мы должны использовать только когда в теле запроса два и более объектов или вообще всегда когда  content: multipart/form-data?

Reply:
Когда multipart/form-data, но можно также использовать RequestParam, но лучше всё же RequestPart

Reply:
Вот в таком эндпойнте например можно же обращаться к параметру просто через @RequestParam или тоже через @RequestPart надо?

/users/me/image:
    patch:
      tags:
        - Пользователи
      summary: 'Обновление аватара авторизованного пользователя'
      operationId: updateUserImage
      requestBody:
        content:
          multipart/form-data:
            schema:
              required:
                - image
              type: object
              properties:
                image:
                  type: string
                  format: binary
      responses:
        '200':
          description: OK
        '401':
          description: Unauthorized

Reply:
Можно и так, и так